src/routing_table/prefix.rs

```diff
@@ -53,36 +53,36 @@ impl<T: Clone + Copy + Default + Binary + Xorable> Prefix<T> {
         if self.bit_count > 0 {
             self.bit_count -= 1;
             // unused bits should be zero:
-            self.name = self.name.with_bit(self.bit_count, false);
+            self.name = self.name.with_bit(self.bit_count(), false);
         }
         self
     }
 
     /// Returns the number of bits in the prefix.
     pub fn bit_count(&self) -> usize {
-        self.bit_count
+        self.bit_count as usize
     }
```

Comment:
This function looks like a good candidate for inlining - it might give us a small performance gain.